### PR TITLE
docs(driver-definitions): Update URL resolver docs

### PR DESCRIPTION
**Pull request by Josmithr**
This PR is an early work in progress. Any recommendations for how to fill in TODOs / expand on what I've already added would be greatly appreciated 🙂

### CODE REVIEW

File: packages/common/driver-definitions/src/urlResolver.ts [hunk 1]
```diff
@@ -4,36 +4,80 @@
...
 	type: "fluid";
+	
 	/**
-	 * The id of the container this resolved url is for.
+	 * The ID of the Container this resolved URL is associated with.
 	 */
 	id: string;
+	
+	/**
+	 * TODO
```
Comment (Josmithr) [inline packages/common/driver-definitions/src/urlResolver.ts:54]:
```suggestion
	 * URL for the Fluid resource.
```

File: packages/common/driver-definitions/src/urlResolver.ts [hunk 2]
```diff
@@ -42,18 +86,29 @@ export interface IContainerPackageInfo {
...
 export interface IUrlResolver {
-	// Like DNS should be able to cache resolution requests. Then possibly just have a token provider go and do stuff?
-	// the expiration of it could be relative to the lifetime of the token? Requests after need to refresh?
-	// or do we split the token access from this?
+	/**
+	 * TODO
+	 * 
+	 * @privateRemarks
+	 * 
+	 * Like DNS should be able to cache resolution requests. Then possibly just have a token provider go and do stuff?
```
Comment (Josmithr) [inline packages/common/driver-definitions/src/urlResolver.ts:98]:
```suggestion
	 * Like DNS, this should be able to cache resolution requests. Then possibly just have a token provider go and do stuff?
```

File: packages/common/driver-definitions/src/urlResolver.ts [hunk 3]
```diff
@@ -42,18 +86,29 @@ export interface IContainerPackageInfo {
...
 
+/**
+ * TODO
+ */
 export interface IUrlResolver {
-	// Like DNS should be able to cache resolution requests. Then possibly just have a token provider go and do stuff?
-	// the expiration of it could be relative to the lifetime of the token? Requests after need to refresh?
-	// or do we split the token access from this?
+	/**
+	 * TODO
```
Comment (Josmithr) [inline packages/common/driver-definitions/src/urlResolver.ts:94]:
```suggestion
	 * Resolves the provided request to a Fluid resource URL.
```

File: packages/common/driver-definitions/src/urlResolver.ts [hunk 4]
```diff
@@ -83,20 +139,37 @@ export interface DriverPreCheckInfo {
  * Additional key in the loader request header
  */
 export enum DriverHeader {
-	// Key to indicate whether the request for summarizer
+	/**
+	 * Key to indicate whether the request for summarizer
```
Comment (Josmithr) [inline packages/common/driver-definitions/src/urlResolver.ts:143]:
```suggestion
	 * Key to indicate whether or not the request is for the summarizer client.
```

File: packages/common/driver-definitions/src/urlResolver.ts [hunk 5]
```diff
@@ -4,36 +4,80 @@
  */
 import { IRequest } from "@fluidframework/core-interfaces";
 
+/**
+ * TODO
+ */
```
Comment (alexvy86) [inline packages/common/driver-definitions/src/urlResolver.ts:9]: A bit of info on this one: 
`IWebResolvedUrl` [is gone](https://github.com/microsoft/FluidFramework/blob/next/BREAKING.md#IResolvedUrl-equivalent-to-IFluidResolvedUrl) and [`IFluidResolvedUrl` is now deprecated](https://github.com/microsoft/FluidFramework/pull/14897), so we're moving towards a world where the only one left is `IResolvedUrl`.

**Pull request closed without merging**